apps/meteor/app/lib/server/functions/setUsername.ts

```diff
@@ -67,78 +68,81 @@ export const setUsernameWithValidation = async (userId: string, username: string
 	void notifyOnUserChange({ clientAction: 'updated', id: user._id, diff: { username } });
 };
 
-export const _setUsername = async function (userId: string, u: string, fullUser: IUser, updater?: Updater<IUser>): Promise<unknown> {
-	const username = u.trim();
+export const _setUsernameWithSession = (session?: ClientSession) =>
+	async function (userId: string, u: string, fullUser: IUser, updater?: Updater<IUser>): Promise<unknown> {
+		const username = u.trim();
 
-	if (!userId || !username) {
-		return false;
-	}
-
-	if (!validateUsername(username)) {
-		return false;
-	}
+		if (!userId || !username) {
+			return false;
+		}
 
-	const user = fullUser || (await Users.findOneById(userId));
-	// User already has desired username, return
-	if (user.username === username) {
-		return user;
-	}
-	const previousUsername = user.username;
-	// Check username availability or if the user already owns a different casing of the name
-	if (!previousUsername || !(username.toLowerCase() === previousUsername.toLowerCase())) {
-		if (!(await checkUsernameAvailability(username))) {
+		if (!validateUsername(username)) {
 			return false;
 		}
-	}
-	// If first time setting username, send Enrollment Email
-	try {
-		if (!previousUsername && user.emails && user.emails.length > 0 && settings.get('Accounts_Enrollment_Email')) {
-			setImmediate(() => {
-				Accounts.sendEnrollmentEmail(user._id);
-			});
+
+		const user = fullUser || (await Users.findOneById(userId));
```

Comment:
pass session

Reply:
This is another place where the user collection shouldn't have been written to yet. I can pass the session here too.